activemq-broker/src/main/java/org/apache/activemq/broker/region/DestinationStatistics.java

```diff
@@ -252,7 +255,22 @@ public void setParent(DestinationStatistics parent) {
             maxUncommittedExceededCount.setParent(null);
             networkEnqueues.setParent(null);
             networkDequeues.setParent(null);
+            // [AMQ-9437] Advanced Message Statistics do not parent.
         }
     }
 
+    public synchronized void setAdvancedMessageStatisticsEnabled(boolean advancedMessageStatisticsEnabled) {
```

Comment:
There is a lot of thread safety issues in this entire class. By making all of this stuff protected someone could easily screw up the state outside a lock. Furthermore you are synchronizing this method but not other places like reset() or anywhere else.

You should not be allowing the ability to set messageFlowStats or to set advancedMessageStatisticsEnabled() outside this method as you could get them out of sync and right now it's easy to do because they are not private

I need to think about it more but I really don't think the current approach is great, especially if you are making member variables protected and not private.